python/src/aac/parser.py

```diff
@@ -8,11 +8,105 @@
 import os
 
 import yaml
+
 from attr import Factory, attrib, attrs, validators
 from yaml.parser import ParserError as YAMLParserError
 
+from aac.util import is_path_name
+
+
+@attrs
+class SourceLocation:
+    """The position and ... of an AaC structure in the source.
+
+    Attributes:
+        line (int): The line number on which the object was found.
+        column (int): The character position at which the object was found.
+        position (int): The position relative to the start of the file where the object was found.
+        span (int): The number of characters occupied by the object relative to `position`.
+    """
+
+    line: int = attrib(validator=validators.instance_of(int))
+    column: int = attrib(validator=validators.instance_of(int))
+    position: int = attrib(validator=validators.instance_of(int))
+    span: int = attrib(validator=validators.instance_of(int))
+
+
+@attrs
+class Lexeme:
+    """A lexical unit for a parsed AaC object.
```

Comment:
If I'm not mistaken, then this might clarify the summary a little.
```suggestion
    """A lexical unit for a parsed AaC definition.
```